server/src/main/java/org/apache/iotdb/db/metadata/MManager.java

```diff
@@ -647,45 +651,75 @@ public void deleteStorageGroups(List<String> storageGroups) throws MetadataExcep
   /**
    * Check if the given path is storage group or not.
    *
-   * @param path Format: root.node.(node)*
+   * @param detachedPath Format: [root, node]
    * @apiNote :for cluster
    */
-  boolean isStorageGroup(String path) {
+  boolean isStorageGroup(List<String> detachedPath) {
     lock.readLock().lock();
     try {
-      return mtree.isStorageGroup(path);
+      return mtree.isStorageGroup(detachedPath);
     } finally {
       lock.readLock().unlock();
     }
   }
 
+
   /**
    * Get series type for given seriesPath.
    *
-   * @param path full path
+   * @param detachedPath detachedPath of full path
    */
-  public TSDataType getSeriesType(String path) throws MetadataException {
+  public TSDataType getSeriesType(List<String> detachedPath) throws MetadataException {
     lock.readLock().lock();
     try {
-      if (path.equals(SQLConstant.RESERVED_TIME)) {
+      if (detachedPath.get(0).equals(SQLConstant.RESERVED_TIME)) {
         return TSDataType.INT64;
       }
 
-      return mtree.getSchema(path).getType();
+      return mtree.getSchema(detachedPath).getType();
+    } finally {
+      lock.readLock().unlock();
+    }
+  }
+
+  /**
+   * Get series type for given measurementMNode
+   *
+   * @param measurementMNode measurementMNode
+   */
+  public TSDataType getSeriesTypeByMNode(MeasurementMNode measurementMNode) {
```

Comment:
Is it necessary to lock the mmanager readlock? or just lock the parent lock of the measurementMNode.

Reply:
fixed.